drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -549,39 +563,25 @@ def introspect_custom_data_element_definitions(set_definition_identifier: false)
 
     cded_records = []
     walk_definition_nodes do |item|
-      # skip unless custom_field_key specified for this item
+      # Skip non-questions items (Groups and Display items)
```

Comment:
I think the comments here are tripping me up a bit --
- line 566 says "skip non-question items", but line 567 skips _unless_ it's a non-question item, maybe this is a typo?
- line 572 says "skip items that already map to a custom data element," but I think what this line is actually doing is skipping items that don't specify a custom mapping in the json definition. Then on line 576 `cdeds_by_key[key]` that's where it is finding out whether this item already maps to an existing CDED. Am I understanding that correctly?

Reply:
good catch, thank you! updated to `next if` instead of `next unless`. This function (`introspect_custom_data_element_definitions`) does not have any tests and likely won't be used at all going forward, but I think it might make sense to keep it around for a bit in case its helpful for debugging issues